cpp/src/arrow/compute/kernels/scalar_temporal.cc

```diff
@@ -364,29 +379,68 @@ struct ISOYear {
 // ----------------------------------------------------------------------
 // Extract ISO week from temporal types
```

Comment:
These comments might need updating (at least 'Extract week from temporal types', the definition of ISO week can be left here)

Reply:
I've just removed "ISO". Do you thing a bigger change is needed?

Reply:
This is good, thanks.